### PR TITLE
Provide out of the box quickstart configuration

### PR DESCRIPTION
**Pull request by bersace**
This PR provides:

- Make config file relocatable by resolving relative path from config file directory
- Update samples to be runnable as a quickstart setup
- Installs sample in `share/temboard-agent/quickstart/`

## Tests instructions

- Create a fresh debian e.g. jessie or stretch with postgresql
- Mount temboard-agent source tree
- Just run `sudo -u postgres ./temboard-agent -c share/temboard-agent.conf`
- Test the API with e.g. `curl -k -X POST -d '{"username": "alice", "password": "alice"}' https://127.0.0.1:2345/login`
- Install with e.g. `pip install .`
- Move out of source tree.
- Run `sudo -u postgres temboard-agent -c /usr/local/share/temboard-agent/quickstart/temboard-agent.conf`
- Test the REST API like above.

### CODE REVIEW

File: share/users [hunk 1]
```diff
@@ -0,0 +1,2 @@
+alice:Xm4iJH8aQmXRvrsLy/fNbQU23K6/FlEVRhhPZKO2aHvYVGzCft+u7RXQ9Z48IyVtnS+n47UM3QUM4is0BVTPOA==
+bob:f9Yol14q9N89EyYSCUR9izXfXugNiXN/HiKsa8CFnIzq/MADFgL7UnjMnxBVypSP6vzPbusotyEAF5H6tP2Mlw==
```
Comment (julmon) [inline share/users]: There are serious security concerns about providing default users. Please remove it.

File: share/temboard-agent.conf [hunk 2]
```diff
@@ -4,16 +4,17 @@ port = 2345
...
 # SSL: certificat file path (.pem)
-ssl_cert_file = /etc/temboard-agent/ssl/temboard-agent_CHANGEME.pem
+ssl_cert_file = temboard-agent_CHANGEME.pem
 # SSL: private key file path (.key)
-ssl_key_file = /etc/temboard-agent/ssl/temboard-agent_CHANGEME.key
+ssl_key_file = temboard-agent_CHANGEME.key
 # Data directory for internal usage.
 # Each running agent *must* has its own and dedicated home directory.
-home = /var/lib/temboard-agent/main
+home = /var/lib/postgresql
```
Comment (julmon) [inline share/temboard-agent.conf]: This path does not exist on some OS like centos/RHEL

Comment (bersace) [inline share/temboard-agent.conf]: Ok, let's use an environment variable.

File: share/users (hunk 1 above)
Comment (bersace) [inline share/users]: This is not default users, this is *quickstart* users. Can we quickstart without users ?

Comment (bersace) [inline share/users]: The file is 0644 by default. Let's see how to manage this.

File: README.rst [hunk 3]
```diff
@@ -29,7 +29,7 @@ http://temboard.io/ for the big picture.
 .. code-block::
 
    pip install temboard-agent
-
+   sudo -u postgres temboard-agent -c /usr/local/share/temboard-agent/quickstart/temboard-agent.conf
```
Comment (pgiraud) [inline README.rst]: Is this path correct?
Shouldn't this be `.../quickstart/share/...`?

File: temboardagent/configuration.py [hunk 4]
```diff
@@ -1,22 +1,29 @@
+import logging
+
 try:
     import configparser
 except ImportError:
     import ConfigParser as configparser
 
-import os
+import os.path
```
Comment (pgiraud) [inline temboardagent/configuration.py]: Is there any reason why you have to import `os.path`?

Comment (bersace) [inline temboardagent/configuration.py]: hmm, actually, os.path is a submodule of `os`. It happen to be imported already, even in a simple `python` interactive shell. But it's not theorically right. See:

```
>>> import logging
>>> logging.handlers
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'handlers'
>>> import logging.handlers
>>> logging.handlers
<module 'logging.handlers' from '/usr/lib/python2.7/logging/handlers.pyc'>
>>>
```

The submodules `handlers` does not exists until it is explicitly imported. Thus, if use the `os.path` module, i import it explicitly.

The other point is that `import X.Y` creates actually *one* variable in module scope, named `X`. `Y` is an attribute of `X` pointing to the submodule. This mean that importing `X.Y` is equivalent to `import X, X.Y`. This is why i *replace* `import os` with `import os.path`.

But yes, that's useless. :-) I remember a case where `os` was imported but *not* `os.path`, but i can't reproduce it.

File: README.rst (hunk 3 above)
Comment (bersace) [inline README.rst]: It is mapped to `data_files` from `setup.py`:

``` python
setup(
    # ...
    data_files=[
        ('share/temboard-agent/quickstart/', [
            'share/temboard-agent.conf',
            'share/temboard-agent_CHANGEME.pem',
            'share/temboard-agent_CHANGEME.key',
            'share/temboard-agent_ca_certs_CHANGEME.pem',
            'share/temboard-agent.logrotate'
        ]),
        ('lib/systemd/system', ['temboard-agent.service']),
    ])
```

Do you have a different installation ?

Comment (pgiraud) [inline README.rst]: Sorry my mistake.

**Comment by bersace** [comment]
I would like to merge this before #42 to ease testing .deb package out of the box.

**Comment by bersace** [comment]
Let's move on.

**Pull request merged**